js/exporter/jspdf/export_data_grid_2.js

```diff
@@ -1,90 +1,155 @@
 import { isDefined } from '../../core/utils/type';
 
+class PdfTable {
+    constructor(drawTableBorder, rect) {
+        this.drawTableBorder = drawTableBorder;
+        this.rect = rect;
+        this.rows = [];
+    }
+
+    addRow(cells) {
+        if(!isDefined(cells)) {
+            throw 'cells is required';
+        }
+        this.rows.push(cells);
+        for(let i = 0; i < cells.length; i++) {
+            const currentCell = cells[i];
+            if(currentCell.drawLeftBorder === false) {
+                if(i >= 1) {
+                    cells[i - 1].drawRightBorder = 0;
+                }
+            } else if(!isDefined(currentCell.drawLeftBorder)) {
+                if(i >= 1 && cells[i - 1].drawRightBorder === false) {
+                    currentCell.drawLeftBorder = false;
+                }
+            }
+
+            if(currentCell.drawTopBorder === false) {
+                if(this.rows.length >= 2) {
+                    this.rows[this.rows.length - 2][i].drawBottomBorder = false;
+                }
+            } else if(!isDefined(currentCell.drawTopBorder)) {
+                if(this.rows.length >= 2 && this.rows[this.rows.length - 2][i].drawBottomBorder === false) {
+                    currentCell.drawTopBorder = false;
+                }
+            }
+        }
+    }
+
+    drawTo(doc) {
+        drawTable(doc, this);
+    }
+}
+
+class PdfGrid {
+    constructor(splitByColumns) {
+        this._splitByColumns = splitByColumns ?? [];
```

Comment:
imho we can remove splitByColumns from ctor and keep in startNewTable only

Reply:
This '_splitByColumns' property is used in two methods: 'startNewTable' and 'addRow', at the moment it is not clear how to replace it in the 'addRow' method.